core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -902,11 +917,8 @@ protected final void doStop() {
                     }
                 }
 
-                for (Iterator<NodeChannels> it = connectedNodes.values().iterator(); it.hasNext();) {
-                    NodeChannels nodeChannels = it.next();
-                    it.remove();
-                    IOUtils.closeWhileHandlingException(nodeChannels);
-                }
+                IOUtils.closeWhileHandlingException(() -> IOUtils.closeWhileHandlingException(connectedNodes.values()),
+                    () -> IOUtils.closeWhileHandlingException(openConnections), openConnections::clear, connectedNodes::clear);
```

Comment:
It is reasonably challenging to figure out what is going on here. Maybe like a comment about the order of operations or something here?

I think it's closing nodes. Followed by closing any "light" channels. And then clearing the maps?